pkg/auth/handler/disable.go

```diff
@@ -123,9 +123,8 @@ type SetDisableHandler struct {
 func (h SetDisableHandler) ProvideAuthzPolicy() authz.Policy {
 	return policy.AllOf(
 		authz.PolicyFunc(policy.DenyNoAccessKey),
-		authz.PolicyFunc(policy.RequireAuthenticated),
 		authz.PolicyFunc(policy.RequireMasterKey),
-		authz.PolicyFunc(policy.DenyDisabledUser),
+		policy.RequireValidUserFunc,
```

Comment:
I think the original policy here is wrong. It should be either (master key) or (RequireValidUser)